### PR TITLE
Build Wix with Cpack

### LINKED ISSUE #7955: deskflow-core does not ship depends
### Feature Request

Deskflow core needs to ship dependencies

### PR DESCRIPTION
**Pull request by sithlord48**
- Build a wix package with cpack 
- Clean up the deploy folder

fixes #7955

### CODE REVIEW

**Review by nbolton — CHANGES_REQUESTED** [review]
The banner text color has changed from white to black. Can this be changed to white using CPack?

deskflow text color

**Review by nbolton — CHANGES_REQUESTED** [review]
The app name is lower case. Can it be changed to title case?
`deskflow` -> `Deskflow`
deskflow lower case

**Review by nbolton — CHANGES_REQUESTED** [review]
The app no longer auto-starts as it did before. 

Auto start is fairly standard with Windows apps since the start menu is a bit of a pain on Windows. Alternatively, creating a desktop shortcut is usually the next best thing (for the same reason). 

Windows is the only OS we tend to do this on, due to its usability limitations.

File: deploy/dist/wix/Product.wxs [hunk 1]
```diff
@@ -1,152 +0,0 @@
...
-
-				<RegistryKey Id="$(var.AppId)" Root="HKLM"
-					Key="Software\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers"
-					ForceCreateOnInstall="yes" ForceDeleteOnUninstall="yes">
-					<RegistryValue Type="string" Name="[INSTALLFOLDER]$(var.GuiBin)" Value="~ HIGHDPIAWARE WIN7RTM"/>
-				</RegistryKey>
-				<!-- Windows 8 and later only -->
-				<Condition><![CDATA[Installed OR (VersionNT >= 602)]]></Condition>
-			</Component>
-			<Merge Id="VC_Redist" SourceFile="$(var.DeployResDir)\dist\wix\msm\Microsoft_VC142_CRT_x64.msm" DiskId="1" Language="0"/>
```
Comment (nbolton) [inline deploy/dist/wix/Product.wxs:40]: Removing this causes:
windows missing dep

**Review by nbolton — CHANGES_REQUESTED** [review]

**Review by nbolton — CHANGES_REQUESTED** [review]
Can we change the license text size? It's pretty hard to read.

win11 text size

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -241,15 +241,11 @@ jobs:
...
-          python ./scripts/package.py --package-version ${{env.DESKFLOW_PACKAGE_VERSION}}
-          mv dist/deskflow* build/
-        env:
-          WINDOWS_PFX_CERTIFICATE: ${{ secrets.WINDOWS_PFX }}
-          WINDOWS_PFX_PASSWORD: ${{ secrets.WINDOWS_PFX_PASS }}
+# When we are ready to try signing on windows
+#      - name: Sign Package
+#        if: ${{ runner.os == 'Windows' }}
+#        run: |
+#          signtool sign -f ${{ secrets.WINDOWS_PFX }}  -p '${{ secrets.WINDOWS_PFX_PASS }}' -t http://timestamp.digicert.com -fd SHA256 ${{github.workspace}}/build/deskflow-*.msi
```
[the remark this answers is not in the document]
Comment (sithlord48) [reply .github/workflows/continuous-integration.yml:248]: sure i just didn't want to forget it , this way all we need todo is remove the comments when we our repo is setup

File: .github/actions/install-dependencies/action.yml [hunk 3]
```diff
@@ -95,6 +95,8 @@ runs:
     - name: Install Wix
       if: ${{ runner.os == 'Windows' }}
       run: |
-        dotnet tool install --global wix --version 4.0.4
-        wix extension add --global WixToolset.UI.wixext/4.0.4
+        dotnet tool install --global wix --version 5.0.2
+        wix extension add --global WixToolset.UI.wixext/5.0.2
+        wix extension add --global WixToolset.Util.wixext/5.0.2
+        wix extension add --global WixToolset.Firewall.wixext/5.0.2
```
[the remark this answers is not in the document]
Comment (sithlord48) [reply .github/actions/install-dependencies/action.yml:101]: Done (added to wiki) these are already used

File: deploy/dist/wix/Product.wxs (hunk 1 above)
Comment (sithlord48) [reply deploy/dist/wix/Product.wxs:40]: Should be resolved now, Tested latest installer in a clean win10 VM and it runs after installing.

**Comment by sithlord48** [comment]
> The app no longer auto-starts as it did before.
> 
> Auto start is fairly standard with Windows apps since the start menu is a bit of a pain on Windows. Alternatively, creating a desktop shortcut is usually the next best thing (for the same reason).
> 
> Windows is the only OS we tend to do this on, due to its usability limitations.

This is  due to the missing dlls , should work now

File: .github/workflows/continuous-integration.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nbolton) [reply .github/workflows/continuous-integration.yml]: If it was a new draft PR, I think it'd be less likely to be forgotten.

**Review by nbolton — CHANGES_REQUESTED** [review]
Maybe I missed this on the last pass. The path shouldn't have the version number in it.

df win11 path

**Review by nbolton — CHANGES_REQUESTED** [review]
Hmm, start menu item seems to be missing.

deskflow missing start
deskflow start menu dir

**Review by nbolton — CHANGES_REQUESTED** [review]
Seeing an OpenSSL dir error.

deskflow openssl missing

**Review by nbolton — COMMENTED** [review]
Looks like your firewall exception is not working as intended. The rule added by the MSI has a weird filename.

firewall exception

firewall rules

**Review by nbolton — APPROVED** [review]
Tested on fresh Windows 11 VM, works great. 

7th time's a charm :wink:

**Pull request merged**